### DISCUSSION TITLE
How to hide / get rid of something from a module

### DISCUSSION THREAD

**Discussion by koksikus** [directus/directus#10598] [Archive]
How to hide / get rid of what I marked with the arrow. As a result, the visitor can see all users from different roles. Better if he only previewed what is in his role folder ...
And how to hide other users from it. Now, when I uncheck the ability to read their data, there is no permission to enter the collections. 
![Zrzut ekranu 2021-12-18 115205](https://user-images.githubusercontent.com/90528895/146638458-691a2bdd-51ad-45f6-9623-251abdec2dcb.jpg)

And the second, just like in the first case, with the difference that it shows no files, but just click on one folder of the list and they all display great
![Zrzut ekranu 2021-12-18 115016](https://user-images.githubusercontent.com/90528895/146638664-fe08aeb4-3fd3-40ec-90cf-8efb4ec67be0.jpg)
.

**Comment by rijkvanzanten** [comment]
There's no way to configure those as of right now

**Reply by koksikus** [reply]
Thanks for the answer.  I can always count on you.  Maybe it would be nice if you could choose the folder to be displayed after selecting the module (in settings).
Now the visitor can see even the manager's accounts (who have this account).  It's like a data leak on people shouldn't know...

**Reply by rijkvanzanten** [reply]
> Now the visitor can see even the manager's accounts (who have this account). It's like a data leak on people shouldn't know...

That totally depends on the permissions you've configured for the other role though! You can configure the permissions so users in others roles can only see users from their own role, or even their own user. 🙂

**Reply by koksikus** [reply]
I agree with you, but when I set the permissions appropriately (I think) for the view in the role. I have a correctly displayed role folder but there is still an All User folder and just click on it and the entire user list will be displayed. When I change the settings in users. I do not have access to the item collection but yes it shows the correct user.

**Reply by rijkvanzanten** [reply]
> I have a correctly displayed role folder but there is still an All User folder and just click on it and the entire user list will be displayed.

The users that are returned are controlled by the permissions configured for Directus Users. In this case, you might want to set that to `{ "role": "$CURRENT_ROLE" }` to only allow users to see other users in the same role, or `{ "id": "$CURRENT_USER" }` to only allow the user to view their own profile.

**Reply by koksikus** [reply]
That's exactly what I do, and as long as the roles work properly (Directus Roles in Read use { "role": "$CURRENT_ROLE" }) and I have still have access to the records in the collection. 

But when I change in Directus Users in Read (use { "id": "$CURRENT_USER" }) Yes, it shows the correct user and only current user but then I enter in collections any record the error "You don't have permission to access this." pops up, I click dismiss twice and then see data in the record.

**Reply by rijkvanzanten** [reply]
> Yes, it shows the correct user and only current user but then I enter in collections any record the error "You don't have permission to access this." pops up, I click dismiss twice and then see data in the record.

Sounds like you're trying to show some user data to the user that doesn't have access to that user data. You might want to disable any user fields (like user_created, etc) you have on your collections from this particular role to ensure they can't accidentally request data they don't have access to.

**Reply by koksikus** [reply]
Thanks. Thanks. Thanks. Thanks. Thanks. Thanks.
You showed me my mistake, I didn't think about the fact that I share user data in the collections!!! (like user_created, etc)

**Reply by rijkvanzanten** [reply]
No problem!

**Reply by koksikus** [reply]
I know why I didn't think of it. Because when I see the entire collection, I see the user created and updated columns (empty - when is set { "id": "$CURRENT_USER" } as we mentioned before) and there no error showing (that is good) only when I enter inside the record, and there no one will see these fields anyway because in the settings it is set "hidden in detailis" by default, an error appears (which we mentioned earlier).

**Reply by rijkvanzanten** [reply]
That error is actually something we could clean up a bit. Instead of showing the big dialog that blocks the user, we could also show that error inline on the m2o field itself, so it shows a notice box with something like "You don't have access to this" 🤔

**Reply by koksikus** [reply]
Well, there can't be empty field as in the columns of the collection (created and updated columns when is set { "id": "$CURRENT_USER" }) because the user will think that this is a systemic error, so such a message "You don't have access to this" is the best solution. 
Now it was enough to check if the field is set hidden in the detail so it does not appear so it does not have to pop up an error.

To sum up the folders, my first thing - "File library" shows photos but not those in folders - maybe it should display all files which the user has permissions. This is the case with folder All Users - which can display all users... These are different folder behaviors, but they should do the same, perhaps I do not know your arrangements for folder settings and you decided that this is the best way. I think "All files" does what "File library" can do...

Thank you for the help!!!!

**Reply by rijkvanzanten** [reply]
> To sum up the folders, my first thing - "File library" shows photos but not those in folders - maybe it should display all files which the user has permissions. This is the case with folder All Users - which can display all users... These are different folder behaviors, but they should do the same, perhaps I do not know your arrangements for folder settings and you decided that this is the best way. I think "All files" does what "File library" can do...

I disagree. The "File Library" folder at the top is your root folder. Any folder you have in that root folder will show up as folders. Files that are in a nested folder don't show up in the root folder as well, this mirrors most modern file system explorers (like Finder on mac / explorer on windows).

**Reply by koksikus** [reply]
Ok, I understand, but at the moment I am greeted by the message "no files", which is not entirely true. Maybe folders should appear instead of the message or massage "Files in folders" or set that after clicking on a module, it goes to the folder selected in some settings... 
Just an idea. I do not insist. 
Directus is Great!!!!
